python/graphstorm/eval/evaluator.py

```diff
@@ -936,255 +719,70 @@ class GSgnnLPEvaluator():
-        self._val_perf_rank_list.append(val_score)
-        return rank
-
-    @property
-    def target_nidx(self):
-        """ target_nidx
-        """
-        return self._target_nidx
-
-    @property
-    def metric(self):
-        """ evaluation metrics
-        """
-        return self._metric
-
-    @property
-    def best_val_score(self):
-        """ Best validation score
-        """
-        return self._best_val_score
-
-    @property
-    def best_test_score(self):
-        """ Best test score
-        """
-        return self._best_test_score
-
-    @property
-    def best_iter_num(self):
-        """ Best iteration number
-        """
-        return self._best_iter
-
-    @property
-    def val_perf_rank_list(self):
-        """ validation performance rank list
-        """
-        return self._val_perf_rank_list
-
-    @property
-    def eval_frequency(self):
-        """ Evaluation frequency.
-        """
-        return self._eval_frequency
-
-    @property
-    def task_tracker(self):
-        """ Task tracker of this evaluator
+        val_mrr: float
+            Validation mrr score
+        test_mrr: float
+            Test mrr score
         """
-        return self.tracker
-
-class GSgnnMrrLPEvaluator(GSgnnLPEvaluator):
-    """ The class for link prediction evaluation using Mrr metric.
+        with th.no_grad():
+            if test_scores is not None:
+                test_score = self.compute_score(test_scores)
+            else:
+                for metric in self.metric_list:
+                    test_score = {metric: "N/A"} # Dummy
 
-    Parameters
-    ----------
-    eval_frequency: int
-        The frequency (# of iterations) of doing evaluation.
-    data: GSgnnEdgeData
-        The processed dataset
-    num_negative_edges_eval: int
-        Number of negative edges sampled for each positive edge in evalation.
-    lp_decoder_type: str
-        Link prediction decoder type.
-    use_early_stop: bool
-        Set true to use early stop.
-    early_stop_burnin_rounds: int
-        Burn-in rounds before start checking for the early stop condition.
-    early_stop_rounds: int
-        The number of rounds for validation scores used to decide early stop.
-    early_stop_strategy: str
-        The early stop strategy. GraphStorm supports two strategies:
-        1) consecutive_increase and 2) average_increase.
-    """
-    def __init__(self, eval_frequency, data,
-                 num_negative_edges_eval, lp_decoder_type,
-                 use_early_stop=False,
-                 early_stop_burnin_rounds=0,
-                 early_stop_rounds=3,
-                 early_stop_strategy=EARLY_STOP_AVERAGE_INCREASE_STRATEGY):
-        eval_metric = ["mrr"]
-        super(GSgnnMrrLPEvaluator, self).__init__(eval_frequency,
-            eval_metric, use_early_stop, early_stop_burnin_rounds,
-            early_stop_rounds, early_stop_strategy)
-        self.train_idxs = data.train_idxs
```

Comment:
Are train_idxs used in this class?

Reply:
No. I look through the calling line, and found this `train_idxs` and `num_negative_endge_eval` were coming from original initial M5 code. And they were set in other place, never used in Evaluator class.